src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -358,21 +358,33 @@ public async Task<IActionResult> SetDeploymentTarget(string sessionId, [FromBody
 
             var serviceProvider = CreateSessionServiceProvider(state);
             var orchestrator = CreateOrchestrator(state, serviceProvider);
+            var cloudApplicationNameGenerator = serviceProvider.GetRequiredService<ICloudApplicationNameGenerator>();
 
-            if(!string.IsNullOrEmpty(input.NewDeploymentRecipeId) &&
-               !string.IsNullOrEmpty(input.NewDeploymentName))
+            if (!string.IsNullOrEmpty(input.NewDeploymentRecipeId))
             {
+                var newDeploymentName = input.NewDeploymentName ?? string.Empty;
+
                 state.SelectedRecommendation = state.NewRecommendations?.FirstOrDefault(x => string.Equals(input.NewDeploymentRecipeId, x.Recipe.Id));
-                if(state.SelectedRecommendation == null)
+                if (state.SelectedRecommendation == null)
                 {
                     return NotFound($"Recommendation {input.NewDeploymentRecipeId} not found.");
                 }
 
-                state.ApplicationDetails.Name = input.NewDeploymentName;
+                if (state.SelectedRecommendation.Recipe.DeploymentType != Common.Recipes.DeploymentTypes.ElasticContainerRegistryImage)
+                {
+                    // We only validate the name when the recipe deployment type is not ElasticContainerRegistryImage.
+                    // This is because pushing images to ECR does not need a cloud application name.
```

Comment:
Doesn't `ApplyAllReplacementTokens` reuse the `cloudApplicationName` as the ECR Repository Name?

Is it possible by skipping the validation here, we're setting an invalid ECR Repository Name?

Reply:
ECR Repository names have their own [validators](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes/DeploymentBundleDefinitions/Container.deploymentbundle#L33). Even if an invalid value is set, it will be flagged by the option setting validator.

Reply:
Is that being run for server mode? In the current public build of the VS Toolkit, I am able to initiate a deployment with an invalid ECR Repository Name that I set via the _"Application name"_ field, that results in a deployment failure:

![image](https://user-images.githubusercontent.com/1170880/167660268-ad727f88-fee2-401a-be32-8775310d147f.png)

Reply:
I cannot set any application name while selecting the "Push Container Images to Amazon Elastic Container Registry" deployment recipe. That field is blocked for me.

Reply:
We discussed internally that we are running the option validator in this workflow, but the regex it's validating does not match what the service is actually enforcing.

Submitted this separately as https://github.com/aws/aws-dotnet-deploy/issues/549, but okay proceeding here.